corehq/apps/userreports/pillow.py

```diff
@@ -218,8 +218,7 @@ def _rebuild_sql_tables(self, adapters):
             for table_name in tables_to_act_on.rebuild:
                 sql_adapter = table_map[table_name]
                 pillow_logging.info(
-                    "[rebuild] Rebuilding table: %s, from config %s at rev %s",
-                    (table_name, sql_adapter.config._id, sql_adapter.config._rev)
+                    f"[rebuild] Rebuilding table: {table_name}, from config {sql_adapter.config._id} at rev {sql_adapter.config._rev}"
```

Comment:
The problem with the non-f-string version was that there was only a single string formatting parameter (a tuple containing the intended three parameters) passed into the logging function. It could have been fixed like so:
```py
                pillow_logging.info(
                    "[rebuild] Rebuilding table: %s, from config %s at rev %s",
                    table_name, sql_adapter.config._id, sql_adapter.config._rev
                )
```

While it probably doesn't matter here, in cases where string formatting has significant computational overhead it is better to use `%s`-style formatting since the logging system defers formatting until it knows that the log message actually needs to be written. This allows expensive debug logging to remain in the code and yet easily disabled via logging configuration in non-debug environments. F-strings are always evaluated (of course) and therefore the computational overhead cannot be avoided via logging config.

Reply:
Oh that's interesting @millerdev, I'd never noticed this feature of the logging functions. Thanks for the explanation